src/cocoa/BugsnagCocos2dxPlugin.mm

```diff
@@ -0,0 +1,58 @@
+#import "BugsnagCocos2dxPlugin.h"
+
+static NSString *const BUGSNAG_COCOS2DX_VERSION = @"1.1.0";
+
+namespace cocos2d {
+    extern const char* cocos2dVersion();
+}
+
+@interface BugsnagClient ()
+- (void)addRuntimeVersionInfo:(NSString *)info withKey:(NSString *)key;
+@end
+
+@interface BugsnagCocos2dxPlugin () <BugsnagPlugin>
+@end
+
+@implementation BugsnagCocos2dxPlugin
+
++ (NSString *)parseCocos2dVersion:(const char *)rawVersion {
+    // cocos2d::cocos2dVersion() is in the format "cocos2d-x-3.1.0" or
+    // "cocos2d-x 3.17.0", so we make some small adjustments to correctly format
+    // the value to a semantic version.
+    NSString *ns_rawVersion =
+        rawVersion == NULL ? nil
+                           : [NSString stringWithCString:rawVersion
+                                                encoding:NSUTF8StringEncoding];
+    NSCharacterSet *separators =
+        [NSCharacterSet characterSetWithCharactersInString:@"- "];
+    return [[ns_rawVersion
+        componentsSeparatedByCharactersInSet:separators] lastObject];
+}
+
+- (void)load:(BugsnagClient * _Nonnull)client {
+    NSString *cocos2dVersion = [BugsnagCocos2dxPlugin parseCocos2dVersion:cocos2d::cocos2dVersion()];
+    [client addRuntimeVersionInfo:cocos2dVersion withKey:@"cocos2dx"];
+    [client addRuntimeVersionInfo:BUGSNAG_COCOS2DX_VERSION withKey:@"Bugsnag Cocos2d-x"];
+}
+
+- (void)unload {
+    // Nothing to do
+}
+
++ (BugsnagClient *_Nonnull)start {
+    BugsnagConfiguration *configuration = [BugsnagConfiguration loadConfig];
+    return [self startWithConfiguration:configuration];
+}
+
++ (BugsnagClient *_Nonnull)startWithApiKey:(NSString *_Nonnull)apiKey {
+    BugsnagConfiguration *configuration = [BugsnagConfiguration loadConfig];
+    configuration.apiKey = apiKey;
+    return [self startWithConfiguration:configuration];
+}
+
++ (BugsnagClient *_Nonnull)startWithConfiguration:(BugsnagConfiguration *_Nonnull)configuration {
+    [configuration addPlugin:[[BugsnagCocos2dxPlugin alloc] init]];
+    return [Bugsnag startWithConfiguration:configuration];
+}
```

Comment:
These seem over-the-top when consumers could instead call `addPlugin:` themselves - this wasn't the case in the past and was the reasoning for having the `+[BugsnagCocos2dxPlugin register...:]` API. Expected usage would be something like:

```objc
BugsnagConfiguration *configuration = [BugsnagConfiguration loadConfig];
[configuration addPlugin:[BugsnagCocos2dxPlugin new]];
[Bugsnag startWithConfiguration:configuration];
```

Reply:
@kstenerud – I think we'd previously agreed not to have breaking changes as part of this. Is there something about the new Cocoa library that makes the current syntax not work?

Reply:
Hey @kstenerud I think we were missing some discussion here :point_up:

Reply:
The old approach used a BugsnagCocosdxPlugin class method to add the plugin, but this is not how the plugin system works anymore, and keeping the old API as-is would interfere with the configuration API.

Trying to keep the old API and also support config is what led down the road to the above wrapping API in the first place. But in the end with the old API unusable, it's better to just adopt the new plugin API and use all the parts as designed.

Reply:
That makes sense - it helps to state the rationale somewhere so everybody is on the same page, especially where the implementation and the original plan differed.

Reply:
The current state doesn't change from the current plan. The previous three method API addition did. Since it's been removed, I didn't see any point in talking about it.

Reply:
The plan didn't discuss changing initialization and its clear we had different ideas about how to overcome API updates breaking the old flow.